app/tests/evaluation_tests/test_admin.py

```diff
@@ -1,17 +1,38 @@
 import pytest
 
 from grandchallenge.evaluation.admin import PhaseAdmin
+from grandchallenge.evaluation.utils import SubmissionKindChoices
 from tests.components_tests.factories import ComponentInterfaceFactory
+from tests.evaluation_tests.factories import PhaseFactory
+from tests.factories import ChallengeFactory
 
 
 @pytest.mark.django_db
 def test_disjoint_interfaces():
     i = ComponentInterfaceFactory()
+    p = PhaseFactory(challenge=ChallengeFactory())
     form = PhaseAdmin.form(
-        data={"algorithm_inputs": [i.pk], "algorithm_outputs": [i.pk]}
+        instance=p,
+        data={"algorithm_inputs": [i.pk], "algorithm_outputs": [i.pk]},
     )
     assert form.is_valid() is False
     assert (
         "The sets of Algorithm Inputs and Algorithm Outputs must be unique"
         in str(form.errors)
     )
+
+
+@pytest.mark.django_db
+def test_read_only_fields_disabled():
+    p1, p2 = PhaseFactory.create_batch(
+        2,
+        submission_kind=SubmissionKindChoices.ALGORITHM,
+        challenge=ChallengeFactory(),
+    )
+    p1.parent = p2
+    p1.save()
+    form = PhaseAdmin.form(
+        instance=p1,
+    )
+    for field in p1.read_only_fields_for_dependent_phases:
```

Comment:
The test is a little bit circular. If `read_only_fields_for_dependent_phases` returns no data, nothing will be tested. I think it's more robust to be explicit about what fields you expect to be disabled in this scenario.

Reply:
Yeah ok, I understand. My thinking was that since I'm testing the specific fields in another test, it would suffice to only test here that any fields in `read_only_fields_for_dependent_phases` - if any - are disabled. If I check here for specific fields, I'd be testing two things in one test. But I agree that the test is more robust if I'm explicit about the fields.

Reply:
Ah okay, I think I missed the test for explicit fields. Still, I think it's better for tests to be explicit about the asserts. Generally, I think that something is often off with the test design if the asserts are in for loops or if statements which can lead to the assert not being hit.